config/src/main/java/com/typesafe/config/impl/ConfigImpl.java

```diff
@@ -360,6 +361,71 @@ public static void reloadEnvVariablesConfig() {
         EnvVariablesHolder.envVariables = loadEnvVariables();
     }
 
+    private static AbstractConfigObject loadEnvVariablesOverrides() {
+        Map<String, String> env = new HashMap(System.getenv());
```

Comment:
I don't mean to bike shed -- especially as I've seen you've written tests against this. I just though I'd add a suggestion which could (1) perhaps make the different override cases more easily scannable/readable and (2) make the test cases easier:

Instead of conflating the 'System.getenv()' call within the same method which does the override logic, you might consider having something like:

static String envVariableAsProperty(envVariable : String, envVarOverridePrefix : String) : String = {
    ... your code from line 377 here
}

which you can then just test via:

assertEquals(envVariableAsProperty("PREFIX_A_B__C", "PREFIX_"), "a.b-c")
assertEquals(envVariableAsProperty("PREFIX_A_B___C", "PREFIX_"), "a.b_c")

etc

Reply:
it makes totally sense, thanks for pushing me out from my laziness :-)